src/pages/search/SearchResults.tsx

```diff
@@ -1,68 +1,55 @@
-import React, { useState } from 'react';
+import React, { FC, Fragment } from 'react';
 import { useTranslation } from 'react-i18next';
-import { useDispatch, useSelector } from 'react-redux';
-import { useHistory } from 'react-router-dom';
 import styled from 'styled-components';
-
-import { List, ListItem, ListItemIcon, ListItemText, Typography } from '@material-ui/core';
-import { Pagination } from '@material-ui/lab';
-import ImageIcon from '@material-ui/icons/Image';
-
-import { search } from '../../api/publicationApi';
-import { RootStore } from '../../redux/reducers/rootReducer';
-
-interface SearchResultsProps {
-  publications: any[];
-  searchTerm: string;
-}
+import { Link } from 'react-router-dom';
+import { Link as MuiLink } from '@material-ui/core';
+import { List, Typography } from '@material-ui/core';
+import PublicationListItemComponent from '../dashboard/PublicationListItemComponent';
+import { SearchResult } from '../../types/search.types';
 
 const StyledSearchResults = styled.div`
   padding-bottom: 1rem;
 `;
 
-const SearchResults: React.FC<SearchResultsProps> = ({ publications, searchTerm }) => {
-  const { t } = useTranslation();
-  const dispatch = useDispatch();
-  const history = useHistory();
-  const results = useSelector((state: RootStore) => state.search);
-  const [offset, setOffset] = useState(0);
+interface SearchResultsProps {
+  publications: SearchResult[];
+  searchTerm: string | null;
+}
 
-  const updateSearch = async (offset: number) => {
-    setOffset(offset);
-    if (searchTerm.length) {
-      await search(searchTerm, dispatch, offset);
-      history.push(`/search/${searchTerm}/${offset}`);
-    }
-  };
+const SearchResults: FC<SearchResultsProps> = ({ publications, searchTerm }) => {
```

Comment:
Burde vi ikke bruke samme komponent for å vise søketreff som når vi viser N siste publikasjoner? Antar begge skal vise samme data for hver publikasjon og paginering, bare at vi også vil ha en linje med søketerm øverst hvis det er søketreff

Reply:
kunne kanskje slått sammen til `PublicationList` elns, som bare tar inn en liste med `PublicationListItem` (eller hva det er)

Reply:
ja, det kan være de kan slås sammen mer, men venter med dette til etter paginering. Pagineringen fungerer litt annerledes på søkeapiet: https://unit.atlassian.net/secure/RapidBoard.jspa?rapidView=53&projectKey=NP&modal=detail&selectedIssue=NP-1300

Reply:
litt usikker på om vi egentlig trenger å tenke så mye på paginering mot APIet. Når jeg lagde `UserList` prøvde jeg å dra til med å mocke 10k users i frontend og det virket fortsatt veldig bra med paginering i frontend (vi fant 1k til å være en logisk øvre grense for antall brukere for ntnu/uio).
så hvis man søker etter "ost" og får 500 results (feks), og så kan brukeren bla gjennom disse uten at vi trenger å gjøre nytt kall for hver gang. hvis man ikke finner det man leter etter får man evt søke mer spesifikt med "osteloff" som kanskje bare gir 5 treff.
gir ikke det mening?

Reply:
ok, jo, det høres jo bra ut det da egentlig. Men vi må vel likevel implementere det slik backend har tenkt eller? Må uansett gjøre noe på den paginerings-oppgaven (enten det er å fjerne argument i søkefunksjonen, eller å sette fast øvre grense på antall resultat), så kan ta det da isteden.

Reply:
fast øvre grense tenker jeg ja - så gjør vi paginering i frontend (så lenge det ikke blir for lang ventetid å hente den øvre grensa da)